src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -368,7 +368,24 @@ public abstract class WebRequestPSCmdlet : PSCmdlet, IDisposable
         /// </summary>
         [Parameter]
         [ValidateNotNullOrEmpty]
-        public virtual UnixDomainSocketEndPoint UnixSocket { get; set; }
+        public virtual UnixDomainSocketEndPoint UnixSocket 
+        { 
+            get => _unixSocket; 
+            set
+            {
+                if (Socket.OSSupportsUnixDomainSockets is true)
```

Comment:
Is it possible to create UnixSocket if Socket.OSSupportsUnixDomainSockets is false? I guess no. So it makes no sense to add the check.

Reply:
@iSazonov unfortunately I can't check at the moment, could you test what happens if Socket.OSSupportsUnixDomainSockets is false?

Reply:
UnixDomainSocketEndPoint throws https://source.dot.net/#System.Net.Sockets/System/Net/Sockets/UnixDomainSocketEndPoint.cs,59

Reply:
Actually `System.Net.Sockets.Socket..ctor(AddressFamily, SocketType, ProtocolType)` will throw a `SocketException` with message `An address incompatible with the requested protocol was used`.

Reply:
https://github.com/dotnet/runtime/blob/2c8cb129ed084e374f06eaa29f21e9ab0b9e8066/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs#L81-L100
https://github.com/dotnet/runtime/blob/2c8cb129ed084e374f06eaa29f21e9ab0b9e8066/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs#L39-L57

Reply:
> Is it possible to create UnixSocket if Socket.OSSupportsUnixDomainSockets is false? I guess no. So it makes no sense to add the check.

We probably want to perform the check so that we can create the appropriate `ErrorRecord` with a more informative message.

Reply:
That being said, the error handling of [System.Net.Sockets.SocketPal.CreateSocket](https://github.com/dotnet/runtime/blob/63e39e24b43d9ee0148d39ba23134edc5febf6d4/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs#L39) could be improved.